src/caffe/syncedmem.cpp

```diff
@@ -3,26 +3,41 @@
 #include "caffe/util/math_functions.hpp"
 
 namespace caffe {
+SyncedMemory::SyncedMemory()
```

Comment:
Is there some kind of corresponding test we can substitute for these debugging blocks and all the invocations of `check_device()`? I'd rather do without these.